### PR TITLE
roachtest: add tpcc ldr roachtest

### PR DESCRIPTION
**Pull request by msbutler**
This new roachtest runs a bidirectional replication stream on the tpcc
workload.  At stream start time, the left database has 10 warehouses, which
translates to a 1 GB initial scan, and the right has 1 warehouse. A 500
warehouse workload is run on the databases in steady state for a 10 minutes.

Epic: none

Release note: none

### CODE REVIEW

**Comment by msbutler** [comment]
first 2 commits live on https://github.com/cockroachdb/cockroach/pull/133192

File: pkg/cmd/roachtest/tests/cluster_to_cluster.go [hunk 1]
```diff
@@ -213,18 +213,29 @@ func defaultWorkloadDriver(
...
 }
 
 func (tpcc replicateTPCC) sourceRunCmd(tenantName string, nodes option.NodeListOption) string {
 	// added --tolerate-errors flags to prevent test from flaking due to a transaction retry error
-	return fmt.Sprintf(`./cockroach workload run tpcc --warehouses %d --tolerate-errors {pgurl%s:%s}`,
-		tpcc.warehouses, nodes, tenantName)
+	cmd := roachtestutil.NewCommand(`./cockroach workload run tpcc`).
+		Flag("warehouses", tpcc.warehouses).
+		MaybeFlag(tpcc.duration > 0, "duration", tpcc.duration).
+		Option("tolerate-errors").
```
Comment (dt) [inline pkg/cmd/roachtest/tests/cluster_to_cluster.go:234]: Hm.  I feel like tolerate-errors substantially reduces the correctness checking value of the test. Did we dig into why we thought we needed it here at all? Can't the workload retry retriable errors without it (perhaps with a different flag)?

**Review by dt — APPROVED** [review]

File: pkg/cmd/roachtest/tests/cluster_to_cluster.go (hunk 1 above)
Comment (msbutler) [reply pkg/cmd/roachtest/tests/cluster_to_cluster.go]: Done.

**Comment by msbutler** [comment]
TFTR!

bors r=dt

**Pull request merged**